crates/parser/src/grammar/expressions/atom.rs

```diff
@@ -452,8 +439,21 @@ fn match_arm(p: &mut Parser) -> BlockLike {
     }
     p.expect(T![=>]);
     let blocklike = expr_stmt(p).1;
+
+    // test match_arms_commas
+    // fn foo() {
+    //     match () {
+    //         _ => (),
+    //         _ => {}
+    //         _ => ()
+    //     }
+    // }
+    if blocklike.is_block() {
+        p.eat(T![,]);
+    } else if !p.at(T!['}']) {
+        p.expect(T![,]);
+    }
```

Comment:
Can be simplified to eating , unconditionally, and then failing if not eof or blocklike